backend/src/main/java/com/htecgroup/skynest/service/impl/TagServiceImpl.java

```diff
@@ -119,4 +116,34 @@ public List<TagResponse> getTagsForObject(UUID objectId) {
         .map(e -> modelMapper.map(e, TagResponse.class))
         .collect(Collectors.toList());
   }
+
+    @Override
+    public void untagObject(UUID tagId, UUID objectId) {
+
+      TagEntity tagEntity = tagRepository.findById(tagId).orElseThrow(TagNotFoundException::new);
+      ObjectEntity objectEntity =
+              objectRepository.findById(objectId).orElseThrow(ObjectNotFoundException::new);
+
+      ObjectToTagKey key = new ObjectToTagKey(tagId, objectId);
+      ObjectToTagEntity objectToTagEntity = new ObjectToTagEntity(key, tagEntity, objectEntity);
+
+      if (!objectToTagRepository.existsById(key)) {
+        throw new TagOnObjectNotFound();
+      }
+      if (!tagEntity.getCompany().equals(objectEntity.getCreatedBy().getCompany())) {
```

Comment:
not sure if `CompanyEntity` has an `equals()` method
you could compare by ids?

Reply:
Will check. Will also remove checks here and place them in validation service.